cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -93,6 +118,8 @@ abstract class InstructionNode0 extends Node0Impl {
     // does not use `Instruction.toString` because that's expensive to compute.
     result = instr.getOpcode().toString()
   }
+
+  final override predicate isGLValue() { instr.isGLValue() }
```

Comment:
I expected something like `_ = getInstructionType(_, result)` here, why isn't that necessary? Similarly for the operand case.

Reply:
I _think_ that will be equivalent to this in all cases: `instr.isGLValue()` is implemented as:
```ql
instr.getResultLanguageType().hasType(_, true)
```
and `getInstructionType(instr, result)` is equivalent to:
```ql
Ssa::getResultLanguageType(instr).hasType(result, isGLValue)
```
and you can eye-ball the implementation of `Ssa::getResultLanguageType(instr)` to convince yourself that `Ssa::getResultLanguageType(instr)` returns a glvalue iff `instr.getResultLanguageType()` returns a glvalue.

_However_, I think I like your implementation a lot better since it doesn't rely on the hidden assumption that `Ssa::getResultLanguageType(instr)` and `instr.getResultLanguageType()` agrees on glvalue-ness.

Reply:
b01a45f7be74c94f3f381e08eb50a1066b401fdf changes the implementation to your suggestion. I didn't see any changes to test results.